cssom-view-1/Overview.bs

```diff
@@ -1856,6 +1858,19 @@ The <dfn attribute for=VisualViewport>scale</dfn> attribute must run these steps
 1. If there is no output device, return 1 and abort these steps.
 1. Otherwise, return the <a>visual viewport</a>'s <a>scale factor</a>.
 
+The segments is an array of DOMRects that represent the dimensions of each existing viewport segment.
+Additonal details about the defintion of a viewport segment can be found here : [[css-env-1#viewport-segments]].
+
+The segments attribute must run these steps:
+1. If the <a>visual viewport</a>'s <a for=visualviewport>associated document</a> is not <a>fully active</a>, return 0.
+2. Returns null if called from within an iframe context and abort these steps.
+3. Returns null if there is only a single viewport segment and abort these steps.
+4. Otherwise, return the <a>visual viewport</a>'s segments array.
+
+Note: Based on the data returned for each viewport segment, developers will be able to infer the number of hinges available as well as the hinge orientation.
+The browser window can be moved/resized such that the number of segments and/or their dimensions change.
+In these cases, resize events will fire at which point authors should re-query this property, as the returned FrozenArray is just a snapshot of the current state.
```

Comment:
It'd be good to note here that these are returned in un-zoomed CSS pixels. It's a bit weird that these are part of the VisualViewport API even though they don't have anything to do with it right? In particular, they don't change when zooming and panning, right?

Or do those environment variables change when zooming and panning? That seems unfortunate if so.

Reply:
They don't change when zooming or panning. But unless I don't understand your point, I'm not sure why they have "anything to do" with the viewport? For example, on a symmetrical folding device which is in a folded state, they're helping developers to implement say a split UX/dual pane view in their app/websites which is basically dividing the viewport in half logically and they need the segments to do so. But again, maybe I misinterpreted your comment.

Reply:
@emilio : Take a look [here](https://github.com/foldable-devices/demos). You can see in the pictures what the API is trying to achieve in JS, similar to what the CSS API is doing.